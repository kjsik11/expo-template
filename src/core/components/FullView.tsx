import Styled from "styled-components/native";

const FullView = Styled.View`
  flex: 1;
`;

export default FullView;
