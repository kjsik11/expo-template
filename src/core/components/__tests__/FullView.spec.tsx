import React from "react";
import { Text } from "react-native";
import { render, screen } from "@testing-library/react-native";
import { ReactTestInstance } from "react-test-renderer";

import FullView from "core/components/FullView";

const TEST_ID = "FULL_VIEW";
const TEXT = "FullView Children";
const CHILDREN = <Text>{TEXT}</Text>;

describe("🫀 Core | 🧩 Components | 🗄️ <FullView>", function () {
  let container: ReactTestInstance | null = null;

  beforeEach(function () {
    render(<FullView children={CHILDREN} testID={TEST_ID} />);

    container = screen.getByTestId(TEST_ID);
  });

  it("should render successfully", () => {
    expect(container).toBeDefined();
  });

  it("should render correctly", () => {
    const containerAsJson = screen.toJSON();
    expect(containerAsJson).toMatchSnapshot();
  });

  it("should render with title", () => {
    expect(screen.getByText(TEXT)).toBeTruthy();
  });
});
