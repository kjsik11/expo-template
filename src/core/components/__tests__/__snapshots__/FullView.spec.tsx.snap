// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🫀 Core | 🧩 Components | 🗄️ <FullView> should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
  testID="FULL_VIEW"
>
  <Text>
    FullView Children
  </Text>
</View>
`;
